cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -389,6 +389,9 @@ public void listAppVersions(HttpRequest request, HttpResponder responder,
       @PathParam("namespace-id") final String namespaceId,
       @PathParam("app-id") final String appName) throws Exception {
     validateApplicationId(namespaceId, appName);
+    // enforce LIST privileges on the app
+    accessEnforcer.enforce(new ApplicationId(namespaceId, appName), authenticationContext.getPrincipal(),
```

Comment:
Wondering if it makes sense to check the LIST permission on a single ApplicationId?

I guess it would be something similar to the GET `/apps` list api? https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java#L280

Reply:
Since this is for a single app - it makes sense to use the permission on the app-id I think?

Reply:
I think get StandardPermission.GET on the single app makes sense. 

In a literal sense, this call is listing versions for an app, but in a more conceptual sense it is getting metadata about the app, similar to getting the app detail.